backend/YouTubeV2.Application/Services/UserService.cs

```diff
@@ -67,16 +80,179 @@ public async Task RegisterAsync(RegisterDto registerDto, CancellationToken cance
+        {
+            var callingUser = await GetByIdAsync(callerID);
+            bool getAllData = false;
+
+            User userToGet = callingUser!;
+
+            if (userID.IsNullOrEmpty())
+            {
+                getAllData = true;
+            }
+            else
+            {
+                if (await _userManager.IsInRoleAsync(callingUser!, Role.Administrator))
+                    getAllData = true;
+
+               userToGet = await GetByIdAsync(userID) ?? throw new NotFoundException("User not found");
+            }
+
+            return await GetDTOForUser(userToGet, getAllData, cancellationToken);
+        }
+
+        public async Task EditAsync(string callerID, string userQueryID, UpdateUserDto updateUserDTO, 
+            CancellationToken cancellationToken)
+        {
+            var userToEdit = await VerifyAccessAndGetUserToModify(callerID, userQueryID);
+
+            await _updateUserDTOValidator.ValidateAndThrowAsync(updateUserDTO, cancellationToken);
+            await VerifyEmailUniqueAsync(userToEdit);
+            await VerifyNameUniqueAsync(userToEdit);
+
+            userToEdit.UserName = updateUserDTO.nickname;
+            userToEdit.Name = updateUserDTO.name;
+            userToEdit.Surname = updateUserDTO.surname;
+
+            if (updateUserDTO.avatarImage.IsNullOrEmpty())
+                await _blobImageService.DeleteProfilePictureAsync(userToEdit.Id, cancellationToken);
+            else
+                await _blobImageService.UploadProfilePictureAsync(updateUserDTO.avatarImage!, userToEdit.Id, cancellationToken);
+
+            var currentRole = await GetRoleForUserAsync(userToEdit);
+            var newRole = updateUserDTO.userType;
+
+            if (currentRole.Equals(Role.Creator, StringComparison.InvariantCulture)
+                && newRole.Equals(Role.Simple, StringComparison.InvariantCulture))
+                await ProcessSwitchToSimple(userToEdit, cancellationToken);
+
+            await SwitchUserRoleAsync(userToEdit, currentRole, newRole);
+            
+            await _userManager.UpdateAsync(userToEdit);
+        }
+
+        public async Task DeleteAsync(string callerID, string userID, CancellationToken cancellationToken)
+        {
+            var callingUser = await GetByIdAsync(callerID);
+
+            if (!await _userManager.IsInRoleAsync(callingUser!, Role.Administrator))
+                throw new ForbiddenException("Only ADMIN can delete users");
+
+            var user = await GetByIdAsync(userID) ?? throw new BadRequestException("User not found");
+
+            var userRoles = await _userManager.GetRolesAsync(user);
+            var userClaims = await _userManager.GetClaimsAsync(user);
+
+            if (await _videoService.GetVideoCountAsync(user) > 0)
+                throw new BadRequestException("Cannot delete account - there are still some videos uploaded");
```

Comment:
is it really how is it supposed to work? admin can delete user only if it doesnt have videos?